### PR TITLE
Issue #701: releasenotes builder: release version validation should work on digits

### LINKED ISSUE #701: releasenotes builder: release version validation should work on digits
now our version validation is based on amount of dots (very simple approach that use to work)

as we migrating to 3digits always release number number, see comment at https://github.com/checkstyle/checkstyle/discussions/12436
we need to improve logic and consider that 10.4 version will be 10.4.0.

it will help us naturally bump 10.4.0 to 10.4.1 without any intervention.  That is good for our new more frequent release model and validation of version.

### PR DESCRIPTION
**Pull request by stoyanK7**
Resolves #701

---
For the manual test I modified the script to be pointing to this branch and changed `CS_RELEASE_VERSION` to `10.5.0`. Runs with no errors: https://github.com/stoyanK7/checkstyle/commit/dee3811dd6eb71765ccf3bc085763cfdf876042b

If I change `CS_RELEASE_VERSION` to `10.5.1`, it is failing as expected.
```
[WARN] Issue #11163 "Enforce file size on Java inputs" is not closed! Please review issue https://github.com/checkstyle/checkstyle/issues/11163
[WARN] Issue #11446 "Update Tests to use new 'verifyXxxxxx' method or 'execute' that use inlined config in Input files" is not closed! Please review issue https://github.com/checkstyle/checkstyle/issues/11446
[WARN] Issue #11214 "Specify violation messages in input files." is not closed! Please review issue https://github.com/checkstyle/checkstyle/issues/11214
[WARN] Issue #12330 "Resolve Pitest for Block Profile-2" is not closed! Please review issue https://github.com/checkstyle/checkstyle/issues/12330

Generation ends with 1 errors.

[ERROR] Validation of release number failed. Release number is a patch(10.5.1), but release notes contain 'new' or 'breaking compatability' labels. Please correct release number by running https://github.com/checkstyle/checkstyle/actions/workflows/bump-version-and-update-milestone.yml

```

### CODE REVIEW

File: releasenotes-builder/src/main/java/com/github/checkstyle/MainProcess.java [hunk 1]
```diff
@@ -132,36 +131,35 @@ else if (isMinor(amountOfCommas) && !containsNewOrBreakingCompatabilityLabel) {
...
      *
-     * @param amountOfCommas amount of commas in release number.
-     * @return {@code true} if amount of commas is 1.
+     * @param str string to check.
+     * @return {@code true} if string to check ends with a zero(0).
      */
-    private static boolean isMinor(long amountOfCommas) {
-        return amountOfCommas == 1;
+    private static boolean endsWithZero(String str) {
+        return !str.isEmpty() && str.charAt(str.length() - 1) == '0';
```
Comment (stoyanK7) [inline releasenotes-builder/src/main/java/com/github/checkstyle/MainProcess.java:162]: I know that this method sounds redundant and we could have used built-in `str.endsWith("0")` but Intellij suggests using this instead so I went with it.

File: releasenotes-builder/src/test/resources/com/github/checkstyle/githubPageBreakingCompatibility.txt [hunk 2]
```diff
@@ -1,4 +1,4 @@
-Checkstyle 1.0.0 - https://checkstyle.org/releasenotes.html#Release_1.0.0
+Checkstyle 1.0.1 - https://checkstyle.org/releasenotes.html#Release_1.0.1
```
Comment (romani) [inline releasenotes-builder/src/test/resources/com/github/checkstyle/githubPageBreakingCompatibility.txt:1]: why notes with breacking changes is patch ?
if we do not validate version, lets still keep version reasonable.

**Review by romani — CHANGES_REQUESTED** [review]
items:

File: releasenotes-builder/src/test/resources/com/github/checkstyle/githubPageBreakingCompatibility.txt (hunk 2 above)
Comment (stoyanK7) [reply releasenotes-builder/src/test/resources/com/github/checkstyle/githubPageBreakingCompatibility.txt]: Completely missed this detail. `New` and `breaking changes` are staying minor. Done.

**Review by romani — APPROVED** [review]
Ok to merge

**Comment by rnveach** [comment]
@romani ping!
What is `releasenotes-builder/.attach_pid15148`?

**Comment by romani** [comment]
Removed

**Comment by stoyanK7** [comment]
No idea where those attach_pid files pop from but it happens pretty often and only when working on this project.

**Comment by rnveach** [comment]
@stoyanK7 do you use idea for debugging? https://stackoverflow.com/questions/54598177/in-idea-how-to-auto-remove-attach-pidxxx-file

**Comment by romani** [comment]
we can add it to https://github.com/checkstyle/contribution/blob/master/.gitignore to not bother us next time

**Comment by stoyanK7** [comment]
Yes, I do use IntelliJ IDEA

I see [other projects](https://github.com/spring-projects/spring-security/pull/6860) having it in their `.gitingore`, wouldn't hurt to have it.

**Pull request merged**